tests/integration/nm/checkpoint.py

```diff
@@ -17,79 +17,79 @@
 # along with this program. If not, see <https://www.gnu.org/licenses/>.
 #
 
+from contextlib import contextmanager
 import time
 
-
 import pytest
 
-
 from libnmstate.nm.checkpoint import CheckPoint
 from libnmstate.nm.checkpoint import get_checkpoints
 from libnmstate.nm.checkpoint import NMCheckPointCreationError
 
 
-def test_creating_one_checkpoint():
+@contextmanager
+def _create_cp(ctx, timeout=None, auto_destroy=True):
```

Comment:
Why you are not using the embedded CheckPoint context?

I would recommend using the full name or some other shortcut so it will be clearer.
`_create_checkpoint` or `_create_chkpnt`

Reply:
It's a left over when CheckPoint is not context.

Will restore and rename the function.